### ISSUE TITLE
Allow option to throw Invalid exception as either AsciiTable or dictionary. 

### ISSUE THREAD

**Issue by synchronizing** [guyskk/validr#27]
Validr is currently unsupported for server applications that require verifying POST requests. Since Validr has implemented AsciiTable as opposed to either a dictionary or a list of errors, when creating RESTful API's the thrown exception is absolutely useless.

A better implementation would perhaps be the ability to define some variable within the base Model classes to give better control of the error:

```python
@modelclass
class Model:
    self.error_type = dict 

class Person(Model):
    name=T.str.maxlen(16).desc('at most 16 chars')
    website=T.url.optional.desc('website is optional')
```

This way when an error is found, the server could easily return back what is going on. 

```python
try:
    test = Person(name=True, website='')
except Exception as e:
    return json({'error': 'Invalid key(s) input.', 'keys': e})
```

Where the error `e` would be:

```python
{'name': 'invalid string'}
```

And therefore the server is able to respond:

```python
{
    'error': 'Invalid key(s) input.',
    'keys': {
        'name': 'invalid string'
    }
}
```

For now, unfortunately, I will have to go back to another json verifier. Great project nonetheless!

**Comment by guyskk** [comment]
Good point. I will add some attributes about error datails to Invalid exception.

**Comment by guyskk** [comment]
Fixed in v1.1.1, see also https://github.com/guyskk/validr/wiki/Model-Class#handle-invalid

**Comment by synchronizing** [comment]
Also, thank you!
